### PR TITLE
Fix eventlistener e2e tests broken due to upgrading Triggers to v0.20.1

### PR DESCRIPTION
**Pull request by abayer**
# Changes

After upgrading to Triggers v0.20.1, `TestEventListenerE2E` started failing consistently due to a `CrashLoopBackOff`. After a bunch of debugging, I determined this is due to https://github.com/tektoncd/triggers/pull/1378. When an eventlistener comes up now, it looks for `caBundle` in the appropriate `ClusterInterceptor`, and with how things were setup, we weren't actually creating the core interceptors etc, and the eventlistener SA didn't have permission to even look for `ClusterInterceptor`s at creation time.

To fix this, we need to change how we install Triggers a bit in `test/e2e-common.sh`, and then in `eventListener_test.go`, we need to create the necessary secret and SA _before_ we create the `ClusterRole` and `ClusterRoleBinding` to give that SA the right permissions.

Also, while I was here, I added `t.Helper()` to a bunch of e2e helper functions, and changed cleanup logic to not delete test-created resources/namespaces if an env var is set or the test failed, matching behavior in Pipeline's e2e tests.

/kind misc

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by abayer** [comment]
/hold

Well, that actually made things worse!

**Comment by abayer** [comment]
/hold cancel

cc @tektoncd/cli-maintainers - this will fix the e2e tests that are currently broken on all PRs.

**Review by dibyom — COMMENTED** [review]
/lgtm

**Comment by abayer** [comment]
/retest

Siggggh - it started failing again for new weird reasons as soon as I squashed.

**Comment by abayer** [comment]
Phew, that was just transient noise.

**Review by vinamra28 — COMMENTED** [review]
/meow
/lgtm
/cc @piyush-garg @pradeepitm12 @chmouel

**Review by piyush-garg — APPROVED** [review]

**Pull request merged**
